### PR TITLE
Search template library

### LINKED ISSUE #602: [FEATURE] Search Template Library
**Is your feature request related to a problem? Please describe.**
As a user I want to search the template library

**Tasks**
- Add a basic search which searches the title and description of all templates:
grafik

**Additional context**

### How we code at Formbricks 🤓

- Everything is type-safe
- All UI components are in the package `formbricks/ui`
- Run `pnpm dev` to find a demo app to test in-app surveys at `localhost:3002`
- We use **chatGPT** to help refactor code. Use our [Formbricks ✨ megaprompt ✨](https://github.com/formbricks/formbricks/blob/main/megaprompt.md) to create the right context before you write your prompt.

### PR DESCRIPTION
**Pull request by Meetcpatel**
## What does this PR do?

Fixes #602 

https://drive.google.com/file/d/1NSDGgUmJ5yFWHLfx2QuNY5b1mZjqHbCK/view

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Chore (refactoring code, technical debt, workflow improvements)
- [ ] Enhancement (small improvements)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change adds a new database migration
- [ ] This change requires a documentation update

## How should this be tested?

- [Navigate to create new survey](https://drive.google.com/file/d/1NSDGgUmJ5yFWHLfx2QuNY5b1mZjqHbCK/view)

## Checklist

- [X] Added a screen recording or screenshots to this PR
- [X] Filled out the "How to test" section in this PR
- [X] Read the [contributing guide](https://github.com/formbricks/formbricks/blob/main/CONTRIBUTING.md)
- [X] Self-reviewed my own code
- [X] Commented on my code in hard-to-understand bits
- [X] Ran `pnpm build`
- [X] Checked for warnings, there are none
- [X] Removed all `console.logs`
- [X] Merged the latest changes from main onto my branch with `git pull origin main`
- [X] My changes don't cause any responsiveness issues
- [X] Updated the Formbricks Docs if changes were necessary

### CODE REVIEW

**Comment by jobenjada** [comment]
Hey @Meetcpatel 

Thanks for your work!

Please make sure that:

1. The search bar disables the the category filter so that it always shows the templates from all categories, not only the selected one.
2. Please set the bg of the Search Bar to white. Ideally, you would allow us to pass a className to the Search Bar Component in the same way shadcn does it so that we can custom style it in different places if we need.
3. Increase the default width of the Search Bar by 20%
4. Rework the search functionality with two words:

https://tella.video/johanness-video-gmmm

It's a bit odd that the first search term becomes irrelevant once the user adds a space behind it. 

What I was trying to say is that it would cool if the search doesnt only check for the string "improve trial" but also displays any template which contains "improve" and "trial" anywhere in the description or title but not neccessarily in that order. Does that make sense?

File: packages/ui/components/SearchBox.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import * as React from "react";
+import { cn } from "@formbricks/lib/cn";
+import { SearchIcon } from "./../index";
```
Comment (jobenjada) [inline packages/ui/components/SearchBox.tsx:3]: pls replace with lucide or heroicon :)

**Review by jobenjada — CHANGES_REQUESTED** [review]
Great PR, thanks for the refinement! Almost ready to merge :)

File: packages/ui/components/icons/SearchIcon.tsx
[the remark this answers is not in the document]
Comment (Meetcpatel) [reply packages/ui/components/icons/SearchIcon.tsx:1]: Ok

[the remark this answers is not in the document]
Comment (Meetcpatel) [reply packages/ui/components/icons/SearchIcon.tsx:1]: Done!

File: packages/ui/index.tsx
[the remark this answers is not in the document]
Comment (Meetcpatel) [reply packages/ui/index.tsx:1]: Done!

File: packages/ui/components/SearchBox.tsx (hunk 1 above)
Comment (Meetcpatel) [reply packages/ui/components/SearchBox.tsx]: Done Replaced!

**Comment by Meetcpatel** [comment]
Done! I changed the code.

**Comment by Meetcpatel** [comment]
Hide the clear button for input type "search"

**Review by jobenjada — APPROVED** [review]
great job, thanks for all the tweaking :)

**Pull request merged**